backend/infrahub/graphql/mutations/node_getter/by_hfid.py

```diff
@@ -23,14 +23,56 @@ async def get_node(
         branch: Branch,
         at: str,
     ) -> Optional[Node]:
-        node = None
-        if not node_schema.human_friendly_id or "hfid" not in data:
-            return node
-
-        return await self.node_manager.get_one_by_hfid(
-            db=self.db,
-            hfid=data["hfid"],
-            kind=node_schema.kind,
-            branch=branch,
-            at=at,
-        )
+        if not node_schema.human_friendly_id:
+            return None
+
+        if "hfid" in data:
+            return await self.node_manager.get_one_by_hfid(
+                db=self.db,
+                hfid=data["hfid"],
+                kind=node_schema.kind,
+                branch=branch,
+                at=at,
+            )
+
+        for component in node_schema.human_friendly_id:
+            name = component.split("__")[0]
+            if name not in data.keys():
+                # The update neither includes "hfid" or all components to form an hfid:
+                return None
+
+        hfid: list[str] = []
+        for component in node_schema.human_friendly_id:
+            name, hfid_value = component.split("__", maxsplit=1)
+
+            if name in node_schema.attribute_names:
+                hfid_component = data.get(name).get(hfid_value)
```

Comment:
We should have returned earlier if these keys were missing from data in this section:

```python
        for component in node_schema.human_friendly_id:
            name = component.split("__")[0]
            if name not in data.keys():
                # The update neither includes "hfid" or all components to form an hfid:
                return None
```

But I can update it for clarity.

Reply:
oh I see that now. I think it would be more clear to use `data[name]` if you know the key is present, but that's just my opinion so feel free to ignore